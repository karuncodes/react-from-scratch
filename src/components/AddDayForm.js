import { Component } from 'react'
import { PropTypes } from 'prop-types'

export class AddDayForm extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        const { resort, date, powder, backcountry } = this.props;
        return <form className="add-day-form">
                <label htmlFor="resort"> Resort Name</label>
                <input id="resort" type="text" defaultValue={resort} />

                <label htmlFor="date"> Date</label>
                <input id="date" type="date" defaultValue={date} />

                <label htmlFor="powder"> Powder Day</label>
                <input id="powder" type="checkbox" defaultChecked={powder} />

                <label htmlFor="backcountry"> Backcountry Day</label>
                <input id="backcountry" type="checkbox" defaultChecked={backcountry} />
            </form>
    }
}

AddDayForm.defaultProps = {
    resort: "Park Hotel",
    date: "2018-05-21",
    powder: false,
    backcountry: true
}

AddDayForm.propTypes = {
    resort: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    powder: PropTypes.bool.isRequired,
    backcountry: PropTypes.bool.isRequired
}