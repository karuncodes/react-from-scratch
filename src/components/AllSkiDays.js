import { Component } from 'react'
import { Route } from 'react-router-dom'
import { SkiDayList } from './SkiDayList'

export class AllSkiDays extends Component {
    constructor(props) {
        super(props);
        this.state = {
            allSkiDays: [
                {
                    resort: "Mt Everest",
                    date: new Date("3/2/2018"),
                    powder: true,
                    backcountry: false
                },
                {
                    resort: "hotel kirkwood",
                    date: new Date("4/12/2018"),
                    powder: false,
                    backcountry: false
                },
                {
                    resort: "hotel transylvania",
                    date: new Date("6/1/2018"),
                    powder: true,
                    backcountry: true
                }
            ]
        }
    }
    render() {
        return (
            <div>
                <Route exact path="/list-days" render={props => (
                        <SkiDayList days={this.state.allSkiDays}/>
                )} />
                <Route path="/list-days/:filter" render={({match}) => (
                        <SkiDayList days={this.state.allSkiDays} filter={match.params.filter}/>
                )} />
            </div>
        )
    }
}