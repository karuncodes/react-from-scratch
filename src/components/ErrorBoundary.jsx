import React from 'react'

export class ErrorBoundary extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            hasError: false
        }
    }
    componentDidCatch (error, info) {
        this.setState({hasError: true});
        console.log("component catch", error, info);
    }
    render() {
        if(this.state.hasError) {
            return (<h2> Something went wrong </h2>)
        }
        return this.props.children;
    }
}