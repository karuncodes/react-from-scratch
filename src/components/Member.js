export const Member = ({member}) => (<div>
    <span>admin: {member.admin}</span>
    <span>name: {member.name}</span>
    <img style={{"width": '80px',
                 "height": '80px'}}
         src={member.thumbnail}>{member.name +'s'} Photo</img>
    <span>email: {member.email}</span>
    </div>
)