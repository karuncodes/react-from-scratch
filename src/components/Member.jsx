import React from 'react'
import  { FaShieldAlt }  from 'react-icons/fa'

export const Member = ({admin, name, image, email, makeAdmin, removeAdmin}) => (<div>
    <h1>{name} {(admin)? <FaShieldAlt /> :''}</h1>
    <img src={image} alt="profile picture" />
    <div> email: <a href={`mailto:${email}`}>{email}</a></div>
    {
        (!admin)?
        <a onClick={() => makeAdmin(email)}> Make Admin </a> :
        <a onClick={() => removeAdmin(email)}> Remove Admin </a>

    }
    </div>
)