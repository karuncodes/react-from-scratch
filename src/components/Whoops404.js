export const Whoops404 = () => 
    <div>
        Pages requested not found
    </div>

