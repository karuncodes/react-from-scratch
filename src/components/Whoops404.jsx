//SnapShot testing forces React import else the test fails
import React from 'react'

export const Whoops404 = () =>
    <div>
        Pages requested not found
    </div>