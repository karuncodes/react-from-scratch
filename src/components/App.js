import { Component, Fragment } from 'react'
import { SkiDayCount } from './SkiDayCount'
import { SkiDayList } from './SkiDayList'
import { Whoops404 } from './Whoops404'
import { NavigationMenu } from './NavigationMenu'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
export class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            allSkiDays: [
                {
                    resort: "Mt Everest",
                    date: new Date("3/2/2018"),
                    powder: true,
                    backcountry: false
                },
                {
                    resort: "hotel kirkwood",
                    date: new Date("4/12/2018"),
                    powder: false,
                    backcountry: false
                },
                {
                    resort: "hotel transylvania",
                    date: new Date("6/1/2018"),
                    powder: true,
                    backcountry: true
                }
            ]
        }
    }

    // countDays (filter) {
    //     const { allSkiDays } = this.state
    //     return allSkiDays.filter(
    //         (day) => (filter) ? day[filter] : day ).length
    // }

    render() {
        return  <Fragment>
                    <Switch>
                        <Route exact path="/" render={props => (
                                <SkiDayCount total= {25} />
                        )} />
                        <Route exact path="/list-days" render={props => (
                                <SkiDayList days={this.state.allSkiDays}/>
                        )} />
                        <Route path="/list-days/:filter" render={({match}) => (
                                <SkiDayList days={this.state.allSkiDays} filter={match.params.filter}/>
                        )} />
                        <Route component={Whoops404} />
                    </Switch>
                    <NavigationMenu />
                </Fragment>
    }
}