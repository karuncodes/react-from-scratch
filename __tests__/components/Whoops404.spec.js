import React from 'react'
import { Whoops404 } from '../../src/components/Whoops404'
import renderer from 'react-test-renderer'

it('404 page renders correctly',() => {
    const tree = renderer
                    .create(<Whoops404 />)
                    .toJSON();
    expect(tree).toMatchSnapshot();
})