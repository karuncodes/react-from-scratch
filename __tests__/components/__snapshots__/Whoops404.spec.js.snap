// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page renders correctly 1`] = `
<div>
  Pages requested not found
</div>
`;
